ddtrace/contrib/molten/patch.py

```diff
@@ -0,0 +1,143 @@
+import inspect
+
+import wrapt
+
+import molten
+from ddtrace import Pin, config
+from ddtrace.utils.formats import get_env
+from ddtrace.utils.importlib import func_name
+
+from ...ext import AppTypes
+from ...ext import http
+
+# Configure default configuration
+config._add('molten', dict(
+    service_name=get_env('molten', 'service_name', 'molten'),
+    app='molten',
+    app_type=AppTypes.web,
+    distributed_tracing_enabled=False,
+))
+
+def patch():
+    """Patch the instrumented methods
+    """
+    if getattr(molten, '_datadog_patch', False):
+        return
+    setattr(molten, '_datadog_patch', True)
+
+    pin = Pin(
+        service=config.molten['service_name'],
+        app=config.molten['app'],
+        app_type=config.molten['app_type'],
+    )
+
+    # add pin to module since many classes use __slots__
+    pin.onto(molten)
+
+    _w = wrapt.wrap_function_wrapper
+    _w('molten', 'App.__init__', patch_app_init)
+    _w('molten', 'App.__call__', patch_app_call)
+    _w('molten', 'Router.add_route', patch_add_route)
```

Comment:
We will want a `unpatch` that undoes these, which is mostly useful for tests.

We also need to import/expose `unpatch` in `__init__.py` as well.